xds/src/main/java/io/grpc/xds/XdsClientMetricReporterImpl.java

```diff
@@ -157,19 +167,37 @@ void reportCallbackMetrics(BatchRecorder recorder, XdsClient xdsClient) {
 
   private void computeAndReportResourceCounts(
       Map<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByType,
+      Map<String, String> resourceNameToAuthority,
       MetricReporterCallback callback) {
     for (Map.Entry<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByTypeEntry :
         metadataByType.entrySet()) {
       XdsResourceType<?> type = metadataByTypeEntry.getKey();
+      Map<String, ResourceMetadata> resources = metadataByTypeEntry.getValue();
 
-      Map<String, Long> resourceCountsByState = new HashMap<>();
-      for (ResourceMetadata metadata : metadataByTypeEntry.getValue().values()) {
+      Map<String, Map<String, Long>> resourceCountsByAuthorityAndState = new HashMap<>();
+      for (Map.Entry<String, ResourceMetadata> resourceEntry : resources.entrySet()) {
+        String resourceName = resourceEntry.getKey();
+        ResourceMetadata metadata = resourceEntry.getValue();
+        String authority = resourceNameToAuthority.getOrDefault(resourceName, "");
```

Comment:
This `getOrDefault()` is a bit weird because: 1) it only papers over a bug, since all the resources should be in the map and 2) `""` is an arbitrary choice. It made more sense before you added proper support for `null` authority, but at this point it is misleading. If you wanted to keep it, using `#missing` or `#broken` or some such may be more clear.